enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,438 @@
+into no-op mode if it encounters an unrecognized platform.
+
+#### Cluster Image Registry Operator
+
+For image registry a [storage backend config decision](https://github.com/openshift/cluster-image-registry-operator/blob/99474318db709a6d17d06468b90cdf0dc0fd2b87/pkg/storage/storage.go#L157) is platform specific.
+On a "None" platform type Image Registry Operator goes into no-op state. For unknown platforms the image registry is configured
+with EmptyDir storage. 
+
+Also, Image Registry storage options might be configured to use PVC-backed or external storage systems
+(such as Ceph or S3 compatible object storage) as a day 2 operation.
+
+For now, it seems that no particular action for the "External" platform type is needed within Image Registry Operator,
+since we're providing enough possibilities to customize Image Registry storage backend.
+
+#### Cluster Ingress Operator
+
+Within the ingress operator PlatformType affects two things:
+1. Choosing EndpointPublishingStrategy, which is `HostNetworkStrategyType` for "None" and unknown PlatformType
+2. Creating dns provider on some platforms. This logic does not engage for "None" or unknown PlatformType
+
+Speaking about `LoadBalancer` EndpointPublishingStrategy - Cluster Ingress Operator might be reconfigured for using it
+as a day2 operation. The operator itself creates `Service` objects with correct provider-specific annotations, the actual handling
+of such objects happens in a provider-specific [cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+#### Cluster Network Operator
+
+TBD
+
+### Workflow Description
+
+During [phase 1](#implementation-phases) proposed changes intended to have no effect and the "External"
+platform type should be handled the same as the "None", so no specific user interaction is expected.
+
+### API Extensions
+
+A new optional constant of `PlatformType` type will be added to the "openshift/api".
+
+```go
+const (
+	...
+    // ExternalPlatformType represent generic infrastructure provider. Provider-specific components should be supplemented separately.
+    ExternalPlatformType PlatformType = "External"
+	...
+)
+```
+
+Additionally, the respective external platform spec and status should be added to the infrastructure resource.
+
+```go
+type CloudControllerManagerMode string
+
+const (
+    // Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling MCO to set `--cloud-provider=external` flag to the kubelets.
+    CloudControllerManagerEnabled CloudControllerManagerMode = "Enabled"
+    // No Cloud Controller Manager is expected to be supplied.
+    // Signaling MCO not to set `--cloud-provider` flag to the kubelets.
+    CloudControllerManagerDisabled CloudControllerManagerMode = "Disabled"
+)
+
+// ExternalPlatformSpec holds the desired state for the generic External infrastructure provider.
+type ExternalPlatformSpec struct{
```

Comment:
@lobziik , i was talking with Ju yesterday and she raised a point about external platform users having the ability to specify how their platform is identified. one of the reasons she mentioned is to help identify the various external platforms that we see in the telemetry. currently for platform none this is a an issue that causes to basically assume that all none platform installation are bare metal. i think we should add a string field in the spec that is required to be filled out, it doesn't need to be validated but just something to allow the user to identify the platform.

Reply:
added provider name into status to address this